Feature: Add tile map layer support with support of the z-index property

JumpLink opened the pull request:
This implementation creates layers for each raw layer as suggested in the discussion on #255.

This PR replaces PR #256

eonarheim commented on src/tiled-map-resource.ts:
```diff
@@ -205,10 +208,11 @@ export class TiledMapResource implements Loadable<TiledMap> {
-      scene.add(tm);
+      const tms = this.getTileMapLayers();
+      for (const tm of tms) {
+         const tx = tm.get(TransformComponent);
```
Please remove unused line `const tx = tm.get(TransformComponent);`

eonarheim commented on src/tiled-map-resource.ts:
```diff
@@ -421,28 +423,30 @@ export class TiledMapResource implements Loadable<TiledMap> {
+      for (var rawLayer of this.data.rawMap.layers) {
+         if (rawLayer.type === "tilelayer") {
+            const layer = new TileMap(0, 0, this.data.rawMap.tilewidth, this.data.rawMap.tileheight, this.data.height, this.data.width);
+            const zindex = getProperty<number>(rawLayer.properties, 'zindex')?.value || -1;
```
Now that we have separate layers a good default might be to assign a z in layer order, and use the custom 'zindex' property as an override
```suggestion
            const zindex = getProperty<number>(rawLayer.properties, 'zindex')?.value || layerZIndexBase++;
```

JumpLink replied:
@eonarheim Where do we want to start with the zindex?
Until now, the default value was -1. If we start at -1, then some layers are by default above the hero if no z-index is set.

eonarheim reviewed (changes requested):
@JumpLink This looks great! I have a couple requests before merge.

* Delete unused line
* Assign a default incrementing zindex to each tile layer, if the custom 'zindex' property override isn't present
* Running locally a couple tests are failing, please update those tests so they pass 
   * `npm run test`

![image](https://user-images.githubusercontent.com/612071/141045537-825da433-5d2c-4532-9a63-b6dacb5daaab.png)

JumpLink commented:
@eonarheim Okay I have
* Auto increment zindex for layers
* Fixed the existing tests
* Add new tests to test the zindex on layers
* Updated the example-city.tmx to add a new "above the hero" layer with zindex

eonarheim reviewed (approved):
@JumpLink This is awesome! Thanks!

kamranayub commented:
Wow @JumpLink you are awesome, thanks!
